pkg/debug/start_debug.go

```diff
@@ -0,0 +1,169 @@
+`
+
+var scriptForPodDeletion = `
+kubectl -n %s  wait --for=delete pod/%s --timeout=60s
+`
+
+func StartDebug(operatorNamespace, clusterNamespace, deploymentName, flagValue string) {
+	// Get Kubernetes Client
+	_, clientset, _ := k8sutil.GetKubeClient()
+
+	deployment, deploymentExist := verifyDeploymentExists(clientset, clusterNamespace, deploymentName)
+	if !deploymentExist {
+		os.Exit(1)
+	}
+
+	deploymentSpec := deployment.Spec
+
+	if flagValue != "" {
+		deploymentSpec.Template.Spec.Containers[0].Image = flagValue
+	}
+
+	labels := deployment.Labels
+	labels["ceph.rook.io/do-not-reconcile"] = "true"
+
+	deploymentSpec.Template.Spec.Containers[0].LivenessProbe = nil
+	deploymentSpec.Template.Spec.Containers[0].StartupProbe = nil
+
+	if flagValue != "" {
+		log.Printf("setting debug image to %s\n", flagValue)
+		deploymentSpec.Template.Spec.Containers[0].Image = flagValue
+	}
+
+	fmt.Println("setting debug command to main container")
+
+	deploymentSpec.Template.Spec.Containers[0].Command = []string{"sleep", "infinity"}
+	deploymentSpec.Template.Spec.Containers[0].Args = []string{}
+
+	if err := updateDeployment(clientset, clusterNamespace, deployment); err != nil {
+		os.Exit(1)
+	}
+
+	deployment_Pod_Name, err := podRunningWithLabel(clientset, clusterNamespace, deploymentSpec)
+	if deployment_Pod_Name == "" || err != nil {
+		os.Exit(1)
+	}
+
+	if err := getAndUpdatetheScale(clientset, clusterNamespace, deployment.Name, 0); err != nil {
+		os.Exit(1)
+	}
+
+	fmt.Printf("waiting for the deployment pod %s to be deleted\n", deployment_Pod_Name)
+
+	waitForPodToDeleted := exec.Command("/bin/bash",
+		"-e", // Exit immediately if a command exits with a non-zero status.
+		"-m", // Terminal job control, allows job to be terminated by SIGTERM
+		"-c", // Command to run
+		fmt.Sprintf(scriptForPodDeletion, clusterNamespace, deployment_Pod_Name),
+	)
+	_, err = waitForPodToDeleted.Output()
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	debug_deployment := &appsv1.Deployment{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-debug", deploymentName),
+			Namespace: clusterNamespace,
+			Labels:    labels,
+		},
+		Spec: deploymentSpec,
+	}
+
+	debugDeployment, err := clientset.AppsV1().Deployments(clusterNamespace).Create(context.TODO(), debug_deployment, v1.CreateOptions{})
+	if err != nil {
+		fmt.Printf("Error creating deployment %s. %v\n", debug_deployment, err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("ensure the debug deployment %s is scaled up\n", deploymentName)
+
+	if err := getAndUpdatetheScale(clientset, clusterNamespace, debugDeployment.Name, 1); err != nil {
+		os.Exit(1)
+	}
+
+}
+
+func getAndUpdatetheScale(clientset *k8s.Clientset, clusterNamespace, deploymentName string, scaleCount int) error {
+	scale, err := clientset.AppsV1().Deployments(clusterNamespace).GetScale(context.TODO(), deploymentName, v1.GetOptions{})
+	if err != nil {
+		fmt.Printf(" failed to get scale of deployment %s. %v\n", deploymentName, err)
```

Comment:
Is the leading space intended on some of these messages?

Reply:
no, removed